### ISSUE TITLE
Add reset() function for every property of any Node and for Nodes themselves

### ISSUE THREAD

**Issue by MaaaxiKing** [godotengine/godot-proposals#1022]
**Describe the project you are working on:**

**Describe the problem or limitation you are having in your project:**
If I want to reset the font color of a label because I have changed it in code, I'll have to do add_color_override("font_color", ColorN("white")) for example. In the inspector there's a reset button next to every changed property. Why can't there be such a function?!
**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
Many lines would be really shorter. It's very practical. If you want to reset all properties of a Node you could call this reset function on the node.
**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
text.reset() align.reset()
**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
It will be used often but yes, it can be worked around but this is annoying.
**Is there a reason why this should be core and not an add-on in the asset library?:**

**Comment by Sslaxx** [comment]
As in reset to default value(s)? If so, having it called `reset_to_default` or something similar would be clearer.

**Comment by MaaaxiKing** [comment]
Yes, exactly what I was thinking of.` reset_to_default()` is a nice idea.

**Comment by Calinou** [comment]
Note that custom theme properties are different from node properties. They're handled differently internally, so there would have to be two implementations of this: one for general properties, and one for theme properties.

**Comment by MaaaxiKing** [comment]
Maybe it should just be `default()` because it's easy to understand and shorter?

**Comment by MaaaxiKing** [comment]
There should be get_default() but look [here https://github.com/godotengine/godot-proposals/issues/1024](https://github.com/godotengine/godot-proposals/issues/1024) too.

**Comment by Zylann** [comment]
Note that for regular properties, it may not be possible to write `text.reset()`. `text` evaluates to a `String` which doesnt even know which property it came from.

Besides, the editor already knows the defaults of things (see how the circular arrow button is implemented for the inspector?), so a more plausible API would be `obj.reset_property("property_name")`.

**Comment by MaaaxiKing** [comment]
The last part I don't fully understand but `Node.reset_property("property_name")` is also fine.
